src/transformers/models/sam/modeling_sam.py

```diff
@@ -623,10 +623,15 @@ def _embed_points(self, points: torch.Tensor, labels: torch.Tensor, pad: bool) -
         input_shape = (self.input_image_size, self.input_image_size)
         point_embedding = self.shared_embedding(points, input_shape)
 
-        point_embedding[labels == -1] = 0.0
-        point_embedding[labels == -1] += self.not_a_point_embed.weight
```

Comment:
By the way what's the point of these two steps rather than `point_embedding[labels == -1] = self.not_a_point_embed.weight`?

Reply:
not sure either! Thanks for the catch.

Reply:
Ok I'll simplify then